src/linear_algebra/tensor_algebra.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Adam Topaz.
+-/
+
+import ring_theory.algebra
+import linear_algebra
+
+/-!
+# Tensor Algebras
+
+Given a commutative ring R, and an R-module M, we construct the tensor algebra of M.
+This is the free R-algebra generated by the module M.
+
+## Notation
+
+1. `tensor_algebra R M` is the tensor algebra itself.
+  It is endowed with an R-algebra structure.
+2. `univ R M` is the R-linear map `M → tensor_algebra R M`.
+3. Given a linear map `f : M → A` to an R-algebra A,
+  `lift R M f` is the lift of `f` to an R-algebra morphism
+  `tensor_algebra R M → A`.
+
+## Theorems
+
+1. `univ_comp_lift` states that the composition
+  `(lift R M f) ∘ (univ R M)` is identical to `f`.
+2. `lift_unique` states that whenever an R-algebra morphism
+  `g : tensor_algebra R M → A` is given whose composision with `univ R M` is `f`,
+  then one has `g = lift R M f`.
+-/
+
+universes u1 u2 u3
+
+variables (R : Type u1) [comm_ring R]
+variables (M : Type u2) [add_comm_group M] [module R M]
+
+namespace tensor_algebra
+
+-- This inductive type is used to express representatives
+-- of the tensor algebra.
+@[nolint doc_blame]
+inductive pre
+| of : M → pre
+| of_scalar : R → pre
+| add : pre → pre → pre
+| mul : pre → pre → pre
+
+namespace pre
+
+instance : inhabited (pre R M) := ⟨of_scalar 0⟩
+
+-- These instances are only used to simplify the notation.
+@[nolint doc_blame]
```

Comment:
Can you also add all the nolint attributes in one go, just as you did with the local instance attribute?

Reply:
I don't think we should add any `@[nolint doc_blame]` anywhere in mathlib. The possibility to add this annotation is meant only for transitioning from undocumented mathlib to documented mathlib.

Reply:
No problem. I'll add the docstrings for these definition.